theano/compile/mode.py

```diff
@@ -333,7 +353,7 @@ def requiring(self, *tags):
                                               self.provided_optimizer)
         return self.__class__(linker=link, optimizer=opt.requiring(*tags))
 
-    def clone(self, link_kwargs=None, **kwargs):
+    def clone(self, link_kwargs={}, optimizer=None, **kwargs):
```

Comment:
I just read how Python handles mutable default values for arguments. I think this is a really counter-intuitive and simply dangerous way of handling default values but, alas, those are the rules we must play by.